java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java

```diff
@@ -3114,16 +3114,29 @@ public Void visitClass(ClassTree node, Void p) {
     public static boolean isCompatibleWithSwitchExpression(SwitchTree st) {
         boolean firstCase = true;
         Name leftTreeName = null;
-        for (CaseTree ct : st.getCases()) {
+        int CaseCount =0;
+        List<? extends CaseTree> cases = st.getCases(); 
+        
+        for (CaseTree ct : cases) {
+           CaseCount++;
             List<StatementTree> statements = new ArrayList<>(ct.getStatements());
             switch (statements.size()) {
                 case 0:
                     break;
                 case 1:
                     if (firstCase && leftTreeName == null && statements.get(0).getKind() == Tree.Kind.RETURN) {
                         break;
-                    } else {
-                        return false;
+                    } else if (CaseCount == cases.size() && statements.get(0).getKind() == Tree.Kind.EXPRESSION_STATEMENT) {
+                        StatementTree statementTree = statements.get(0);
+                        JCTree.JCExpressionStatement jceTree = (JCTree.JCExpressionStatement) statementTree;
+                        if (!(jceTree.expr instanceof JCTree.JCAssign)) {
+                            return false;
+                        }
+                        JCTree.JCAssign assignTree = (JCTree.JCAssign) jceTree.expr;
+                       if (firstCase || leftTreeName != null && leftTreeName.contentEquals(((JCTree.JCIdent) assignTree.lhs).name)) {
+                            break;
+                        } 
+                       else return false;
```

Comment:
Maybe, we should use `{}` and should format the `if` statement.

Reply:
had corrected the same in recent commit.